assets/prototype/application/ui/components/layout/eeditorModal/useEditorModalManager.ts

```diff
@@ -31,46 +28,33 @@ const useEditorModalManager = (): EditorStateManager => {
 		});
 	};
 
-	const setEditorId = (editorId: EditorId): void => {
-		dispatch({
-			type: ActionType.SET_EDITOR_ID,
-			editorId,
-		});
-	};
-
-	const setEntityId = (entityId: EntityId): void => {
-		dispatch({
-			type: ActionType.SET_ENTITY_ID,
-			entityId,
-		});
-	};
-
-	const closeEditor = (): void => {
+	const closeEditor = (editorId: EditorId): void => {
 		dispatch({
 			type: ActionType.CLOSE_EDITOR,
+			editorId,
 		});
 	};
 
 	return {
-		...state,
-		setEditorId,
-		setEntityId,
+		editors: state,
 		closeEditor,
 		openEditor,
 	};
 };
 
 const reducer: EditorStateReducer = (state, action) => {
 	const { type, ...rest } = action;
-	const { editorId, entityId } = rest;
+	const { editorId } = rest;
 	switch (type) {
 		case ActionType.OPEN_EDITOR:
-			return { ...state, ...rest };
-		case ActionType.SET_EDITOR_ID:
-			return { ...state, editorId };
-		case ActionType.SET_ENTITY_ID:
-			return { ...state, entityId };
+			return [...state, { isOpen: true, ...rest }];
 		case ActionType.CLOSE_EDITOR:
+			const lastEditorId = path<EditorId>(['editorId'], last<EditorState>(state));
+			if (lastEditorId && lastEditorId === editorId) {
+				return dropLast<EditorState>(1, state);
+			}
+			return state;
+		case ActionType.CLOSE_ALL:
```

Comment:
There seem to be a bit of duplication with `useEditorModalState`, what will happen with the latest?